src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// <see cref="JsonConverter"/> for converting exceptions safely.
+    /// If the exception fails to deserialize, returns an <see cref="ExceptionDeserializationException"/> instead of failing.
+    /// </summary>
+    public class SafeExceptionConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(Exception).IsAssignableFrom(objectType);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            try
+            {
+                return serializer.Deserialize(reader, objectType);
+            }
+            catch
```

Comment:
Seems like this class is really an attempt at adding a 'TryDeserialize' to Json.NET for exception types?

Is it so bad to handle this from the caller?

Reply:
You can't handle it from the caller. This deserialization is done as a part of deserializing a larger object, which fails when this field fails to deserialize.